utils/misc/misc.go

```diff
@@ -274,30 +274,42 @@ func AddFileToZip(zipWriter *zip.Writer, filename string) error {
 
 	writer, err := zipWriter.CreateHeader(header)
 	if err != nil {
-		panic(err)
+		return err
 	}
 	_, err = io.Copy(writer, fileToZip)
 	return err
 }
 
 // UnZipSingleFile unzips zip containing single file into ouputfile path passed
-func UnZipSingleFile(outputfile string, filename string) {
+func UnZipSingleFile(outputFile string, filename string) error {
 	r, err := zip.OpenReader(filename)
 	if err != nil {
-		panic(err)
+		return err
 	}
-	defer r.Close()
-	inputfile := r.File[0]
+	defer func(r *zip.ReadCloser) {
+		err := r.Close()
+		if err != nil {
+			pkgLogger.Errorf("error closing the ReadCloser")
```

Comment:
Shall we mention the error as well
```suggestion
			pkgLogger.Errorf("error closing the ReadCloser: %s", err)
```

Reply:
removed the method